rpc/src/main/scala/org/bitcoins/rpc/client/RpcClient.scala

```diff
@@ -0,0 +1,1114 @@
+  }
+
+  def verifyTxOutProof(
+      proof: MerkleBlock): Future[Vector[DoubleSha256Digest]] = {
+    bitcoindCall[Vector[DoubleSha256Digest]]("verifytxoutproof",
+                                             List(JsString(proof.hex)))
+  }
+
+  def walletLock(): Future[Unit] = {
+    bitcoindCall[Unit]("walletlock")
+  }
+
+  def walletPassphrase(passphrase: String, seconds: Int): Future[Unit] = {
+    bitcoindCall[Unit]("walletpassphrase",
+                       List(JsString(passphrase), JsNumber(seconds)))
+  }
+
+  def walletPassphraseChange(
+      currentPassphrase: String,
+      newPassphrase: String): Future[Unit] = {
+    bitcoindCall[Unit](
+      "walletpassphrasechange",
+      List(JsString(currentPassphrase), JsString(newPassphrase)))
+  }
+
+  private def bitcoindCall[T](
+      command: String,
+      parameters: List[JsValue] = List.empty)(
+      implicit
+      reader: Reads[T]): Future[T] = {
+    val request = buildRequest(instance, command, JsArray(parameters))
+    val responseF = sendRequest(request)
+
+    val payloadF: Future[JsValue] = responseF.flatMap(getPayload)
+
+    payloadF.map { payload =>
+      parseResult((payload \ resultKey).validate[T], payload)
+    }
+  }
+
+  case class RpcError(code: Int, message: String)
+  implicit val rpcErrorReads: Reads[RpcError] = Json.reads[RpcError]
+
+  // Should both logging and throwing be happening?
+  private def parseResult[T](result: JsResult[T], json: JsValue): T = {
+    // First catch errors thrown by calls with Unit as the expected return type (which isn't handled by UnitReads)
+    if (result == JsSuccess(())) {
+      (json \ errorKey).validate[RpcError] match {
+        case err: JsSuccess[RpcError] =>
+          logger.error(s"Error ${err.value.code}: ${err.value.message}")
+          throw new RuntimeException(
+            s"Error ${err.value.code}: ${err.value.message}") // More specific type?
+        case _: JsError =>
+      }
+    }
+
+    result match {
+      case res: JsSuccess[T] => res.value
+      case res: JsError =>
+        (json \ errorKey).validate[RpcError] match {
+          case err: JsSuccess[RpcError] =>
+            logger.error(s"Error ${err.value.code}: ${err.value.message}")
+            throw new RuntimeException(
+              s"Error ${err.value.code}: ${err.value.message}") // More specific type?
+          case _: JsError =>
+            logger.error(JsError.toJson(res).toString())
+            throw new IllegalArgumentException(
+              s"Could not parse JsResult: ${(json \ resultKey).get}")
+        }
+    }
+  }
+
+  private def getPayload(response: HttpResponse): Future[JsValue] = {
+    val payloadF = response.entity.dataBytes.runFold(ByteString(""))(_ ++ _)
+
+    payloadF.map { payload =>
+      Json.parse(payload.decodeString(ByteString.UTF_8))
+    }
+  }
+
+  def sendRequest(req: HttpRequest): Future[HttpResponse] = {
+    Http(m.system).singleRequest(req)
```

Comment:
After we send the request, we need to make sure the `HttpResponse` "id" field is the exact same UUID as we sent in the request. Else we need to throw an exception (we have the wrong request)